### DISCUSSION TITLE
(feat): Feature Ideas & Approach

### DISCUSSION THREAD

**Discussion by aayu5hgit** [aayu5hgit/histograde#7] [General]
We will primarily utilize this discussion to share the **ideas** related to the `targeted features` and work upon the `optimized approach` to achieve it.

**Comment by aayu5hgit** [comment]
|  Previous |  Updated  |
|---|---|
![image](https://github.com/aayu5hgit/histograde/assets/86314754/28b2e75e-9961-4fe6-9cd2-aae3feb1e7c5) | ![image](https://github.com/aayu5hgit/histograde/assets/86314754/01345b30-0229-449f-8ed6-7450962c3ba7)
| Coloured | Grayscale |
| No contrasting method applied | CLAHE Contrasting method applied |
| Less Accurate | More Accurate |
|   |   |

**Comment by aayu5hgit** [comment]
## 🔴 Morphological Image Processing Method for cell detection

**Reply by anujbagad** [reply]
Morphological image processing is a technique used in image processing and computer vision to analyze and manipulate the shape and structure of objects within an image.
Operations include: dilation, erosion, opening, and closing

1. Dilation:

Binary Dilation: In binary image processing, dilation is used to expand the boundaries of objects in a binary image. The center of the structuring element is placed on each pixel of the image, and if at least one pixel under the structuring element is "on" (i.e., has a value of 1), the corresponding pixel in the result is set to "on." Dilation makes objects thicker or larger.
@aayu5hgit This can be compared to Otsu's Thresholding
Grayscale Dilation: In grayscale image processing, dilation is similar to binary dilation but considers pixel intensities. Instead of simply checking if a pixel is "on" or "off," grayscale dilation takes the maximum pixel value under the structuring element.
@aayu5hgit this can be compared to CLAHE Contrasting method
 
2. Erosion:
 Erosion is used to shrink the boundaries of objects. It takes the minimum pixel value under the structuring element. Grayscale erosion can be used for noise reduction and to shrink the bright regions in the image.

3. Opening:
Opening works by first applying erosion to remove small details and then applying dilation to recover the original object sizes that are not removed during erosion

4. Closing:
Closing is a combination of binary dilation followed by binary erosion. It is used to close small gaps between objects in a binary image. Closing works by first applying dilation to close gaps between objects and then applying erosion to remove any small protrusions.

**Comment by anujbagad** [comment]
![image](https://github.com/aayu5hgit/histograde/assets/112265370/c82402cb-fda9-4ec9-a7cf-d113ad5ec200)
Cell detection with MIP

**Reply by aayu5hgit** [reply]
Good job @anujbagad 

We need to improve the `accuracy`, find more **complex methods** to do so because I can see most boundaries are covering the nucleus and not the cytoplasm to be detected as a cell.

**Reply by aayu5hgit** [reply]
@anujbagad the above approach is `binary dilation MIP`

Can you try using the `grayscale dilation MIP` which can help us increase the accuracy I believe as we have already got better results when we process `grayscale` images over `colored` images?

**Comment by MansiBellani** [comment]
Wavelet transform is a mathematical technique used for signal processing and image analysis. It's a method that allows you to represent an image in terms of its frequency components, which can be useful for various image processing tasks, including compression, denoising, feature extraction, and more. 
Wavelet transform has advantages over other image transformation techniques, such as the Fourier transform, because it provides information about both frequency and location. This makes it well-suited for tasks where localized features are important, like image compression and denoising. 
It is a powerful tool in image processing that enables the analysis of an image's frequency content at different scales and orientations, making it valuable for various applications in computer vision and image analysis.
